src/compiler/crystal/syntax/lexer.cr

```diff
@@ -26,6 +26,33 @@ module Crystal
     # Heredocs pushed when found. Should be processed when encountering a newline
     getter heredocs = [] of {Token::DelimiterState, HeredocItem}
 
+    property invisible_loc_pragmas : Hash(Int32, Array(LocPragma))? = nil
+
+    alias LocPragma = LocStackPragma | LocSetPragma
+
+    enum LocStackPragma
```

Comment:
What do you think if we separate this into two types? It can be `LocPushPragma` and `LocPopPragma`. Maybe the code will be even smaller. And the `case` bellow will be "complete" (right now there's a hidden `else`)